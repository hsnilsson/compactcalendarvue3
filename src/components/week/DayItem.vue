<template>
  <li
    class="day"
    :class="{
      firstDay: firstDayOfMonth,
      startOfMonth: startOfMonth,
      endOfMonth: endOfMonth,
      holyday: isHolyDay,
      vacation: isVacation
    }"
  >
    <span>
      {{ day.toLocaleDateString(undefined, { day: '2-digit' }) }}
    </span>
  </li>
</template>
<script>
export default {
  props: {
    day: Date,
    changedMonth: Boolean,
    vacations: {
      type: Object,
      default: () => {}
    },
    holidays: {
      type: Array,
      default: () => []
    }
  },
  computed: {
    isVacation: function(){
      let floatDate = parseFloat(`${this.day.getMonth() + 1}.${this.day.toLocaleDateString(undefined, { day: '2-digit' })}`)

      if (Object.prototype.hasOwnProperty.call(this.vacations, this.day.getFullYear())) {
          return this.vacations[this.day.getFullYear()].includes(floatDate)
      }

      return false

    },
    isHolyDay: function() {
      let floatDate = parseFloat(`${this.day.getMonth() + 1}.${this.day.toLocaleDateString(undefined, { day: '2-digit' })}`)
      return this.holidays.includes(floatDate)
    },
    firstDayOfMonth: function() {
      return this.day.getDate() === 1
    },
    startOfMonth: function() {
      return this.day.getDate() <= 7 && this.changedMonth
    },
    endOfMonth: function() {
      return this.day.getDate() > 20 && this.changedMonth
    },
  },
}
</script>

<style scoped>
.vacation {
  background: lightgrey;
}

.firstDay {
  font-weight: bolder;
  background: linear-gradient(to right, #9198e5, white);
  border-left: solid 1px #9198e5;
}
.startOfMonth {
  border-top: solid 1px #9198e5;
}
.endOfMonth {
  border-bottom: solid 1px #9198e5;
}

.holyday {
  font-weight: bolder;
  color: darkorange;
}
</style>
