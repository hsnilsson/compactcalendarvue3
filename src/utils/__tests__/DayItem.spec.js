import { it, describe, expect } from "vitest";

import { mount } from "@vue/test-utils";
import DayItem from "@/components/week/DayItem.vue";

describe("DayItem.vue", () => {
  it("renders correctly", () => {
    // mount the component
    const wrapper = mount(DayItem, {
      props: {
        day: new Date("December 17, 1995 03:24:00"),
        changedMonth: false,
      },
    });

    // generate a new snapshot of
    // the component's template
    expect(wrapper.element).toMatchSnapshot();
  });
});
