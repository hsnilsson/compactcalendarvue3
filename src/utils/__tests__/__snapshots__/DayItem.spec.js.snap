// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DayItem.vue > renders correctly 1`] = `
<li
  class="day"
  data-v-9f0d1115=""
>
  <span
    data-v-9f0d1115=""
  >
    17
  </span>
</li>
`;
